### DISCUSSION TITLE
Help for ABR and transcoding

### DISCUSSION THREAD

**Discussion by MehdiJafari84** [OvenMediaLabs/OvenMediaEngine#1186] [Help] [answered]
Hello great guys
Thanks OME.

I have a few questions and would be very happy if you could help me.
I want to provide four qualities in the live streaming, the original quality and 360, 480 and 720.
I will provide these four qualities if the bitrate of the stream sent is greater than 720, and if it is lower, I will only provide the lower qualities.
This is my setup but the stream always converts to 4 quality even when I send at 480 bitrate.
please guide me.

```
<OutputProfiles>
	<!-- Enable this configuration if you want to hardware acceleration using GPU -->
	<HardwareAcceleration>false</HardwareAcceleration>
	<OutputProfile>
			<Name>bypass_stream</Name>
			<OutputStreamName>${OriginStreamName}</OutputStreamName>
			<Playlist>
					<Name>Paid</Name>
					<FileName>index</FileName>
					<Options>
							<WebRtcAutoAbr>true</WebRtcAutoAbr>
					</Options>

					<Rendition>
							<Name>Orginal</Name>
							<Video>bypass_video</Video>
							<Audio>bypass_audio</Audio>
					</Rendition>

					<Rendition>
							<Name>480p</Name>
							<Video>480p</Video>
							<Audio>bypass_audio</Audio>
					</Rendition>
					<Rendition>
							<Name>720p</Name>
							<Video>720p</Video>
							<Audio>bypass_audio</Audio>
					</Rendition>
			</Playlist>
			<Encodes>
					<Video>
							<Name>bypass_video</Name>
							<Bypass>true</Bypass>
					</Video>
					<Audio>
							<Name>bypass_audio</Name>
							<Bypass>true</Bypass>
					</Audio>
					<Video>
							<Name>480p</Name>
							<Codec>h264</Codec>
							<Bitrate>1000000</Bitrate>
							<Width>854</Width>
							<BypassIfMatch>
									<Codec>eq</Codec>
									<Width>lte</Width>
									<Height>lte</Height>
									<SAR>eq</SAR>
							</BypassIfMatch>
					</Video>
					<Video>
							<Name>720p</Name>
							<Codec>h264</Codec>
							<Bitrate>2048000</Bitrate>
							<Width>1280</Width>
							<BypassIfMatch>
									<Codec>eq</Codec>
									<Width>lte</Width>
									<Height>lte</Height>
									<SAR>eq</SAR>
							</BypassIfMatch>
					</Video>
			</Encodes>
	</OutputProfile>
</OutputProfiles>
```
Conditional transcoding not working for me.

My second problem is with the resolution of the stream sent from mobile phone with Larix, when I send 1280x720 , it is converted to 1280x2548 and upscale.

The 3th question is what is SAR in BypassIfMatch?

4th question is about OvenPlayer, In your demo player does not show the speed and audio in settings but in v.0.10.26 the player show speed and audio in setting in live streaming, How can I disable and hide them?
 
5th question is about quality name in the player setting, How can change it to 360p, 480p and so on?

I appreciate your help in advance
Thank You
(sorry for poor English)

**Comment by getroot** [comment, marked as answer]
BypassIfMatch is not a function that bypasses the creation of the output track, it is a function that creates the track but bypasses the encoding. So even with BypassIfMatch you will see all renditions in your player. Perhaps the feature you are looking for is currently not provided by OME.

I don't understand your second problem, please create a new issue and upload your server.xml and ovenmediaengine.log files.

SAR is Sample Aspect Ratio. The above setting means that the encoding is bypassed only when the ratio of the input video and the output video are the same. (I'll add that description to the manual later.)

Please post OvenPlayer issues to the OvenPlayer repo.

**Reply by MehdiJafari84** [reply]
Thank you for your response.
BypassIsMatch is a good option, but it would be great if there was a way to make the output conditional or not.
The second problem is about resize video and upscale that I read about it in your doc, I'll Remove Height in BypassIsMatch.

**Comment by gregfr** [comment]
Regarding your 4th and 5th question, there's a separate GitHub for Oven Player: https://github.com/AirenSoft/OvenPlayer
You can customize it to some extends with CSS (it can be as simple as "display: hide") or recompile it to your liking.

Regards

**Reply by MehdiJafari84** [reply]
Thank you very much for your suggestion.
